tests/scripts/rm_hpo_api_tests.sh

```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
 	echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}	
 }
 
+# Post the experiment result to HPO
+# input: Experiment result
+# output: Create the Curl command with given JSON and get the result
+function post_experiment_trial() {
+	exp_result=$1
+
+	content_='curl -H "Content-Type: application/json"'
+	_post_=$(curl -H "Content-Type: application/json" -d "${exp_result}" http://localhost:8085/experiment_trials -w '\n%{http_code}' 2>&1)
+	# Example curl command used to post the experiment result: curl -H "Content-Type: application/json" -d {"id" : null, "trial_number": 0, "trial_result": "success", "result_value_type": "double", "result_value": 98.78, "operation" : "EXP_TRIAL_RESULT"} http://localhost:8085/experiment_trials -w n%{http_code}
+	post_exp_result_cmd="${content_} -d ${exp_result} http://localhost:8085/experiment_trials -w '\n%{http_code}'"
+
+	echo "" | tee -a ${LOG_} ${LOG}
+	echo "Command used to post the experiment result= ${post_exp_result_cmd}" | tee -a ${LOG_} ${LOG}
+	echo "" | tee -a ${LOG_} ${LOG}
+
+	echo "${_post_}" >> ${LOG_} ${LOG}
+
+	http_code=$(tail -n1 <<< "${_post_}")
+	response=$(echo -e "${_post_}" | tail -2 | head -1)
+	echo "Response is ${response}" >> ${LOG_} ${LOG}
+}
+
+# Post same result multiple times for the same id & trial number
+function post_duplicate_exp_result() {
+	# Post a valid experiment to RM-HPO /experiment_trials API.
+	exp="valid-experiment"
+	create_post_exp_json_array ${current_id}
+	post_experiment_json "${rm_hpo_post_experiment_json[$exp]}"
+	trial_num="${response}"
+
+	if [ "${http_code}" == "200" ]; then
+		failed=0
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 5
+
+		# Post a valid experiment result to RM-HPO /experiment_trials API.
+		exp_result="valid-experiment-result"
+		
+		echo -n "Post the experiment result to HPO..."
+		create_post_exp_result_json_array ${current_id} ${trial_num}
+		post_experiment_trial "${rm_hpo_post_exp_result_json[$exp_result]}"
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 5
+
+		# Post the duplicate experiment result to RM-HPO /experiment_trials API.
+		echo -n "Post the same experiment result to HPO again for the same id and trial number..."
+		create_post_exp_result_json_array ${current_id} ${trial_num}
+		post_experiment_trial "${rm_hpo_post_exp_result_json[$exp_result]}"
+
+		actual_result="${http_code}"
+		expected_result_="^4[0-9][0-9]"
+		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+
+		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+	else
+		failed=1
+		expected_result_="200"
```

Comment:
Where are these used? Also there needs to be a message for the failure - "post experiment failed"